src/pages/en/guides/styling.md

```diff
@@ -176,6 +176,166 @@ You can also use the `<link>` element to load a stylesheet on the page. This sho
 
 Because this approach uses the `public/` directory, it skips the normal CSS processing, bundling and optimizations that are provided by Astro. If you need these transformations, use the [Import a Stylesheet](#import-a-local-stylesheet) method above.
 
+## Cascading Order
+
+When evaluating multiple CSS rules that apply to the same element, the browser uses _specificity_ and _order of appearance_ to determine which style to show. If one rule is more _specific_ than another, it will take precedence:
+
+```astro title="MyComponent.astro"
+<style>
+  h1 { color: red }
+  div > h1 {
+    color: purple
+  }
+</style>
+<div>
+  <h1 class="red">
+    This header will be purple!
+  </h1>
+</div>
+```
+
+If two rules have the same specificity, the _order of appearance_ is evaluated, and the last rule will take precedence:
+```astro title="MyComponent.astro"
+<style>
+  h1 { color: purple }
+  h1 { color: red }
+</style>
+<div>
+  <h1 class="red">
+    This header will be red!
+  </h1>
+</div>
+```
+
+Among Astro features, if the specificity of a rule is the same, the resulting styles will appear in this order:
+
+- **`<link>` tags in the head** (lowest precedence)
```

Comment:
This is not Astro-specific, so I think it might make sense to provide an MDN link to Cascading Order and make it a bit more clear that you're "teaching CSS" here, not Astro. And, I *might* even remove these initial examples altogether.

OR, I'd keep the examples, but then make the text more super-Astro specific:

``````suggestion
Astro components will sometimes have to evaluate multiple CSS "instructions" for the same element. For example, your component might import a CSS stylesheet, include its own `<style>` tag, *and* be rendered inside a layout.

When multiple CSS rules apply to the same element, browsers first use _specificity_ and then _order of appearance_ to determine which style to show.

If one rule is more _specific_ than another, no matter when the CSS rule is evaluated, it will take precedence:

```astro title="MyComponent.astro"
<style>
  h1 { color: red }
  div > h1 {
    color: purple
  }
</style>
<div>
  <h1>
    This header will be purple!
  </h1>
</div>
```

If two rules have the same specificity, then the _order of appearance_ is evaluated, and the last rule will take precedence:
```astro title="MyComponent.astro"
<style>
  h1 { color: purple }
  h1 { color: red }
</style>
<div>
  <h1>
    This header will be red!
  </h1>
</div>
```

Astro CSS rules are evaluated in this order of appearance:

- **`<link>` tags in the head** (lowest precedence)
``````
Also, I'm not sure about using `class="red"` because we never define what the red class is? Are we supposed to infer that the class has a styling rule to make *something* red? All text? just h1? The `<style>` tag in the component isn't styling the class, it's styling the element. 

It seems these examples work without the class applied to them, and if we're not actually showing where the class comes from, or that it's even being applied (we don't import a global stylesheet, nor do we show that this component is in a layout that does) then I don't think it's helping explain why the style tag is making the h1 red.

Reply:
Oh no, I accidentally left this in when I copied it from an experimentation - will remove! 🤦